File: performance-metrics-java/src/main/java/com/metrics/performancemetrics/entity/Metric.java
package com.metrics.performancemetrics.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.Instant;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity(name = "metric")
@Data
public class Metric {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;
    @Column(nullable = false)
    private String name;
    @Column(nullable = false)
    private Instant created_at;
    @OneToMany(mappedBy = "metric", cascade = CascadeType.ALL)
    private List<MetricValue> metricValues = new ArrayList<>();
}

File: performance-metrics-android/app/src/main/java/com/metrics/performancemetrics/data/Metric.kt
package com.metrics.performancemetrics.data

import com.google.gson.annotations.SerializedName

data class Metric(@SerializedName("id") val id : Int,
                  @SerializedName("name") val name : String,
                  @SerializedName("created_at") val createdAt : String,
                  @SerializedName("metricValues") val metricValues : ArrayList<MetricValue>
    )